### PR TITLE
Bug/17

### PR DESCRIPTION
**Pull request by stefan-kamsker**

### CODE REVIEW

File: src/Bundle/CollectionFieldBundle/Tests/CollectionFieldTypeTest.php [hunk 1]
```diff
@@ -240,7 +240,7 @@ public function testWritingGraphQLData()
         $admin = new User();
         $admin->setRoles([User::ROLE_PLATFORM_ADMIN]);
         $this->container->get('security.token_storage')->setToken(
-          new UsernamePasswordToken($admin, null, 'main', $admin->getRoles())
+          new UsernamePasswordToken($admin, null, 'api', $admin->getRoles())
```
Comment (franzwilding) [inline src/Bundle/CollectionFieldBundle/Tests/CollectionFieldTypeTest.php]: Revert this change but send csrf_token with requests instead

File: src/Bundle/CoreBundle/Form/FieldableFormBuilder.php [hunk 2]
```diff
@@ -6,23 +6,47 @@
...
+    }
+
+    /**
+     * determine if the current authentication is done via API token or not, disable csrf for API Token
+     *
+     * @param $options
+     *
+     * @return array
+     */
+    public function handleCsrfProtection($options)
```
Comment (franzwilding) [inline src/Bundle/CoreBundle/Form/FieldableFormBuilder.php]: For a cleaner solution, disable csrf protection directly in the form and not in the form builder

File: src/Bundle/CoreBundle/Resources/config/services.core.yml [hunk 3]
```diff
@@ -31,5 +31,5 @@ services:
   # The fieldable form builder allows you to build content and setting forms based on the defined contentType / settingType fields.
   united.cms.fieldable_form_builder:
     class: UnitedCMS\CoreBundle\Form\FieldableFormBuilder
-    arguments: ['@form.factory', '@united.cms.field_type_manager']
+    arguments: ['@form.factory', '@united.cms.field_type_manager', '@security.token_storage']
```
Comment (franzwilding) [inline src/Bundle/CoreBundle/Resources/config/services.core.yml]: After moving the logic from the form builder to the form, we don't need this here anymore

File: src/Bundle/CoreBundle/Security/ApiRequestMatcher.php [hunk 4]
```diff
@@ -9,7 +9,8 @@ class ApiRequestMatcher extends RequestMatcher
 {
 
     public function __construct() {
-        parent::__construct('^/[A-Za-z0-9_-]+/[A-Za-z0-9_-]+/api(/|$)');
+        // old one ^/[A-Za-z0-9_-]+/[A-Za-z0-9_-]+/api(/|$) doesnt match sign controller route
```
Comment (franzwilding) [inline src/Bundle/CoreBundle/Security/ApiRequestMatcher.php]: Remove comment

File: src/Bundle/StorageBundle/Resources/config/services.yml [hunk 5]
```diff
@@ -28,4 +28,9 @@ services:
     UnitedCMS\StorageBundle\Command\StorageFileDeleteUnused:
         arguments: ['@doctrine.orm.default_entity_manager', '@united.cms.field_type_manager']
         tags:
-            - { name: 'console.command', command: 'united:storage:delete-unused' }
\ No newline at end of file
+            - { name: 'console.command', command: 'united:storage:delete-unused' }
+
+    # A service for S3 presigning from type
```
Comment (franzwilding) [inline src/Bundle/StorageBundle/Resources/config/services.yml]: Comment is a bit misleading. This is not a service FOR forms, this is a form-as-a-serivce.

File: src/Bundle/CollectionFieldBundle/Tests/CollectionFieldTypeTest.php (hunk 1 above)
Comment (stefan-kamsker) [reply src/Bundle/CollectionFieldBundle/Tests/CollectionFieldTypeTest.php]: not fixed: created an issue for this #13
